playbooks/common/openshift-cluster/initialize_facts.yml

```diff
@@ -8,3 +8,5 @@
       role: common
       local_facts:
         hostname: "{{ openshift_hostname | default(None) }}"
+  - set_fact:
+      openshift_docker_hosted_registry_network: "{{ hostvars[groups.oo_first_master.0].openshift.common.portal_net }}"
```

Comment:
So, I think this is tricky...  ideally anything we would want to coordinate at that level would be managed by a shared/common role that is required by all of the other roles involved...  We've had a bit of dysfunction around defining what that role was in the past, generally falling on openshift_facts, since openshift_common (at least in the past) pulled in deps that we weren't comfortable pulling in for auxiliary roles. 

In reality, it should probably be openshift_common that holds this centralized logic/validation for this type of stuff with a more specific role handling pulling in shared binaries for master, node, etc.

I really hate the idea of managing this type of complexity at the playbook level, since I think it actually makes it more difficult to enforce. That said, if the problem is solely that the vars are being set at too narrow of a scope, then we should address that and do what we can to set them at the proper scope.

As an additional aside, I think evaluate_groups is probably a candidate for re-evaluation in the not too distant future, as we split out the community supported bits into openshift-ansible-contrib and better defining the model for using openshift-ansible from Ansible and/or Ansible Tower directly. It served the purpose of bringing together the bin/cluster playbooks with the byo playbooks, but I think in the long run it's use adds complexity for very little real benefit, especially if we can better document and define the pure-ansible management model.